.github/workflows/pre-commit-workflow.yml

```diff
@@ -0,0 +1,27 @@
+name: pre-commit
+
+on: [push, pull_request]
```

Comment:
What other options are there?

Reply:
https://docs.github.com/en/actions/reference/events-that-trigger-workflows#webhook-events
The ones I added are the only relevant ones for us on workflow. @ahmedhamodi